File: src/plugin/inter/WorldMapInterfacePlugin.java
package plugin.inter;

import org.redrune.game.entity.actor.mask.Animation;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class WorldMapInterfacePlugin extends InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		if (componentId == 44) {
			player.setNextAnimation(new Animation(-1));
		}
		player.getPackets().sendWindowsPane(player.getInterfaceManager().hasRezizableScreen() ? 746 : 548, 2);
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(755);
	}
}

File: src/main/kotlin/plugin/rsinterface/WorldMapInterfacePlugin.kt
package plugin.rsinterface

import org.redrune.game.content.plugin.type.InterfacePlugin
import org.redrune.game.entity.actor.mask.Animation
import org.redrune.game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/31/2017
 */
class WorldMapInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int,
    ): Boolean {
        if (componentId == 44) {
            player.nextAnimation = Animation(-1)
        }
        player.packets.sendWindowsPane(if (player.interfaceManager.hasRezizableScreen()) 746 else 548, 2)
        return true
    }

    override fun register() {
        registerInterfacePlugin(755)
    }
}